proposed/2021/VulnerabilityInfoInPMUI.md

```diff
@@ -0,0 +1,52 @@
+# Expose package vulnerability information in the Package Manager UI - Technical Spec
+
+* Status: **In Review**
+* Author(s): [Jean-Pierre Briedé](https://github.com/jebriede)
+* Issue: [10981](https://github.com/NuGet/Home/issues/10981) Expose package vulnerabilities information in the PMUI
+* Type: Feature
+
+## Problem Background
+
+Developers that install NuGet packages through Visual Studio may be installing packages that have known vulnerabilities or they may have a project that is depending on a vulnerable version of a NuGet package. There is currently no information in the Package Manager UI that indicates that a package version is vulnerable so developers may continue to use a vulnerable package in their project. Users should be able to see a warning about a package version they are browsing or one they have installed so they can be informed of the vulnerability and hopefully update to a different version of the package that is not vulnerable or use a different package.
+
+## Who are the customers
+
+* All users that install a NuGet packages using the Package Manager in Visual Studio.
+* All users that manage their installed NuGet packages using the Package Manager in Visual Studio.
+
+## Requirements
+
+* A warning icon will be shown in the Installed tab and in the package list if the project has any vulnerable packages installed. These icons will be the same ones that are used when deprecated packages are installed. In the case where a package version is both vulnerable and deprecated, only one icon will still be shown, but the tooltip on the icon will reflect the vulnerability and deprecation.
+* Maintain the functionality of the deprecated warning icon in the package list as well as the installed tab. The icon will be used to indicate vulnerabilities, but will need to still indicate deprecated packages.
+* There can be one or more vulnerabilities for a package version so the details pane will need to be able to display a list of vulnerabilities, the severity levels, and the advisory links for each.
+
+## Goals
+
+* Show a warning icon in the Installed tab with a tooltip indicates how many vulnerable packages are installed.
+* Show a warning icon next to the package version of a vulnerable package in the package list. This icon will show for packages in the Browse and Installed tabs.
```

Comment:
```suggestion
* Show the existing warning icon (used for deprecation) next to the package version of a vulnerable package in the package list. This icon will show for packages in the Browse and Installed tabs.
```

Reply:
> This icon will show for packages in the Browse and Installed tabs.

Why not Updates/Consolidate? I believe the XAML is shared, so will this be explicitly coded as a condition?

Reply:
> Looks good, @jebriede --checking on something: seeing the vulnerability information when clicking a package in the browse pane is to cover the fact that a registration blob needs to be loaded explicitly for this detail?

Thanks Drew! It's partially because of where the data is queried from but also because the details pane is where more information is available about a package so the added details about the vulnerability will appear there as well.

Reply:
> > This icon will show for packages in the Browse and Installed tabs.
> 
> Why not Updates/Consolidate? I believe the XAML is shared, so will this be explicitly coded as a condition?

@donnie-msft: Good catch! It should show up there too. I'll remove this sentence altogether.